test/fw/ptl/lib/ptl_server.py

```diff
@@ -1860,7 +1860,14 @@ def create_moms(self, name=None, attrib=None, num=1, delall=True,
             except PbsManagerError as e:
                 rc = e.rc
             if rc:
-                if len(self.status(NODE)) > 0:
+                node_length = 0
+                try:
+                    node_length = len(self.status(NODE))
```

Comment:
If pbsnodes doesn't return any nodes, shouldn't status() return [] and not an exception?  Not having any nodes (or queues for that matter) is a valid PBS configuration.

Reply:
If node is exist but PTL method "create_moms()" creates moms after deleting all the existing nodes and checking nodes are deleted. If a node doesn't exist then pbs command "/opt/pbs/bin/pbsnodes -s centos.pbspro.com -v -a"  itself through a error. err: ['/opt/pbs/bin/pbsnodes: Server has no node list']